doc/doxygen/src/move-semantics.md

```diff
@@ -0,0 +1,76 @@
+Semantics of typedefs and moving their values {#typedef-and-move}
+=============================================
+
+This page describes general properties of types defined (`typedef`'d) in RIOT.
+It is mostly aimed at developers who alter or extend RIOT;
+API users can often ignore them if they stick to idiomatic C.
+
+This is being documented for consistency within RIOT,
+and for the benefit of users who use RIOT
+from other languages (such as @rust using-rust "Rust" )
```

Comment:
Is there some rust documentation on movability that this could link to?

Reply:
It's not like Rust does anything "special" that'd be documented; neither does C strictly forbid moving around values. In particular, the Rust compiler will not willy-nilly move data around, but the ways things are used (RAII without guaranteed return placement, destruction by value) that lead to things being moved.

The `Pin<>` documentation is obviously related, I've linked it as "verbose and complicated" in the latest version. The [proposal for guaranteed copy elision](https://github.com/rust-lang/rfcs/pull/2884) is related as well, but it's not clear enough yet whether it even suffices to link it in here.